code/mob/living/critter/ai/flock/flocktasks.dm

```diff
@@ -659,6 +659,10 @@ stare
 
 /datum/aiTask/succeedable/harvest/on_tick()
 	var/obj/item/harvest_target = holder.target
+	if(!harvest_target || !isturf(harvest_target.loc)) //sometimes there are race conditions, and we should deal with that
+		harvest_target = null
```

Comment:
Redundant assignment to null right before a return?

Reply:
not redundant, nulls out harvest_target to trigger failure() on the next evaluation and reset the task.

Reply:
surely harvest_target is reset anyway due to being a local variable?

Reply:
You probably meant to null out `holder.target`, you mutated the value of `harvest_target`

Reply:
oops, that is exactly what I meant to do. willfix.

Reply:
oops, that is exactly what I meant to do. willfix.